Sources/ITwinMobile/ITMAuthorizationClient.swift

```diff
@@ -481,7 +495,13 @@ open class ITMAuthorizationClient: NSObject, AuthorizationClient, OIDAuthStateCh
         raiseOnUserStateChanged()
         completion(nil)
     }
+    
+    open func resumeExternalUserAgentFlow(_ url: URL) -> Bool {
+        // This is only needed by AppAuth before iOS 11, and we require iOS 12.2.
+        return false
+    }
 
+    // MARK: - AuthorizationClient Protocol implementation
     open func getAccessToken(_ completion: @escaping AccessTokenCallback) {
```

Comment:
I thought you were going to change AccessTokenCallback to add an expirationData parameter so that you don't have to call getAccessToken so often.

Reply:
I saved this issue for the end; I wanted to speak with you and @calebmshafer about the localization of this expiration date. Assuming the expiration date is initially an NSDate, we may need to do something like the following, right?:

```Swift
extension NSDate {
    var stringValue: String {
        let formatter = NSDateFormatter()
        formatter.dateFormat = "ourDateFormat"
        return formatter.stringFromDate(self)
    }
}
```

Reply:
I believe you should use [NSISO8601DateFormatter](https://developer.apple.com/documentation/foundation/nsiso8601dateformatter) to convert the NSDate to a string.

Reply:
Since this is an Objective C protocol, just have the protocol take the NSDate object and use NSISO8601DateFormatter internally.

Reply:
In case my above two comments are confusing, what I meant is:

* Add a nullable NSDate* parameter to AccessTokenCallback before the nullable NSError*.
* Inside the imodel02 host code that calls getAccessToken on the AuthorizationClient and waits for the AccessTokenCallback, use NSISO8601DateFormatter to convert the NSDate* object to a string.
* Pass the string to the backend JavaScript.
* In the backend JavaScript, initialize a JavaScript Date object using that string.

Reply:
IIRC, NSISO8601DateFormatter has a problem reading strings from JavaScript, but the strings that it creates work fine in JavaScript. (Something about the time zone.) If I'm not right about that, then a Google Search for ISO 8601 NSDateFormatter should show a StackOverflow hit that provides instructions on making it work without using NSISO8601DateFormatter.

Reply:
Got it. Thank you Travis, your instructions are perfectly clear.